l10n_it_ddt/models/stock.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Apulia Software s.r.l. (http://www.apuliasoftware.it)
+#    @author Francesco Apruzzese <f.apruzzese@apuliasoftware.it>
+#    Copyright 2015 Lorenzo Battistini - Agile Business Group
+#
+#    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+#
+##############################################################################
+
+
+from openerp import fields, models, api
+
+
+class StockPicking(models.Model):
+
+    _inherit = 'stock.picking'
+
+    ddt_ids = fields.Many2many(
+        comodel_name='stock.picking.package.preparation',
+        relation='stock_picking_pack_prepare_rel',
+        column1='stock_picking_id',
+        column2='stock_picking_package_preparation_id',
+        string='DdT',
+        copy=False, )
+
+    @api.multi
+    def write(self, values):
+        if 'move_lines' in values:
+            pack_to_update = self.env['stock.picking.package.preparation']
+            for picking in self:
+                pack_to_update |= picking.ddt_ids
+        res = super(StockPicking, self).write(values)
+        if 'move_lines' in values:
+            if pack_to_update:
+                pack_to_update._update_line_ids()
```

Comment:
One way
    @api.multi
    def write(self, values):
        pack_to_update = None
        if 'move_lines' in values:
            pack_to_update = self.env['stock.picking.package.preparation']
            for picking in self:
                pack_to_update |= picking.ddt_ids
        res = super(StockPicking, self).write(values)
        if pack_to_update:
            pack_to_update._update_line_ids()
        return res

or this better if super call is not influenced by pack_to_update._update_line_ids() call
    @api.multi
    def write(self, values):
        if 'move_lines' in values:
            pack_to_update = self.env['stock.picking.package.preparation']
            for picking in self:
                pack_to_update |= picking.ddt_ids
            pack_to_update._update_line_ids()
        return super(StockPicking, self).write(values)